lib/extractor/layer.ts

```diff
@@ -46,7 +52,9 @@ export async function extractDockerArchive(
 
     tarExtractor.on("error", (error) => reject(error));
 
-    createReadStream(dockerArchiveFilesystemPath).pipe(tarExtractor);
+    createReadStream(dockerArchiveFilesystemPath)
+      .pipe(gunzip())
```

Comment:
We now support gzip compressed layers.

Reply:
Is this going to ungzip gzipped stream and do nothing if it's not gzip??

Reply:
You are correct! Hence the interesting name of the library `gunzip-maybe` (see https://github.com/mafintosh/gunzip-maybe). 

From the project description: `Transform stream that gunzips its input if it is gzipped and just echoes it if not.`